backend/src/near.js

```diff
@@ -34,9 +66,37 @@ const signRemovedValidators = (removedValidators) => {
   }
 };
 
-const getCurrentNodes = (nodes) => {
-  let currentValidators = nodes.current_validators;
-  let nextValidators = nodes.next_validators;
+const fetchValidatorsInfo = async (validators, validatorArray) => {
+  for (let i = 0; i < validators.length; i++) {
+    const { account_id } = validators[i];
+    let validator = validatorArray.get(account_id);
+    let fee = validator?.fee;
+    let delegators = validator?.delegators;
+
+    if (fee) {
+      validators[i].fee = fee;
+    }
+    if (delegators) {
+      validators[i].delegators = delegators;
+    }
+    fee = await nearRpc.callViewMethod(
+      account_id,
+      "get_reward_fee_fraction",
+      {}
+    );
+    delegators = await nearRpc.callViewMethod(
+      account_id,
+      "get_number_of_accounts",
+      {}
+    );
```

Comment:
You have the cache of `validatorArray`, so we only need to move these updates outside of this method to make its execution independent of this `fetchValidatorsInfo` function, and thus not blocking it on awaiting for the values